core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -217,17 +211,41 @@ static Context deleteContext(Map<String, String> config) {
         Context dataContext = dataContext(config);
 
         String codecAsString = config.get(DELETE_AVRO_COMPRESSION);
-        CodecFactory codec = codecAsString != null ? toCodec(codecAsString) : dataContext.codec();
+        String compressionLevel = config.getOrDefault(DELETE_AVRO_COMPRESSION_LEVEL, AVRO_COMPRESSION_LEVEL_DEFAULT);
+        CodecFactory codec = codecAsString != null ? toCodec(codecAsString, compressionLevel) : dataContext.codec();
 
         return new Context(codec);
       }
 
-      private static CodecFactory toCodec(String codecAsString) {
+      private static CodecFactory toCodec(String codecAsString, String compressionLevel) {
+        CodecFactory codecFactory = null;
         try {
-          return CodecName.valueOf(codecAsString.toUpperCase(Locale.ENGLISH)).get();
+          switch (Codec.valueOf(codecAsString.toUpperCase(Locale.ENGLISH))) {
+            case UNCOMPRESSED:
+              codecFactory = CodecFactory.nullCodec();
+              break;
+            case SNAPPY:
+              codecFactory = CodecFactory.snappyCodec();
+              break;
+            case ZSTD:
+              codecFactory = CodecFactory.zstandardCodec(
+                  compressionLevelAsInt(compressionLevel, ZSTD_COMPRESSION_LEVEL_DEFAULT));
+              break;
+            case GZIP:
+              codecFactory = CodecFactory.deflateCodec(
+                  compressionLevelAsInt(compressionLevel, GZIP_COMPRESSION_LEVEL_DEFAULT));
+              break;
+            default:
+              throw new IllegalArgumentException("Unsupported compression codec: " + codecAsString);
+          }
         } catch (IllegalArgumentException e) {
           throw new IllegalArgumentException("Unsupported compression codec: " + codecAsString);
         }
+        return codecFactory;
+      }
+
+      private static int compressionLevelAsInt(String tableCompressionLevel, int defaultCompressionLevel) {
```

Comment:
can use `PropertyUtil.propertyAsInt`, no need for another util method

Reply:
PropertyUtil operates on properties maps (string to string) so we would have to pass that in to the toCodec helper, since for the default value we need to be aware of the specific compression codec.

Reply:
I see, makes sense to me